### ISSUE TITLE
Specific vs. generic identifiers

### ISSUE THREAD

**Issue by nissimsan** [w3c-ccg/traceability-vocab#230]
We recently discussed whether a BOL number is best described with the more general 
`https://schema.org/identifier`
or the more narrow/precise 
`https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Bill_of_lading_number`

Neither is wrong, a BL# is both of these definitions. I can see use cases and arguments for both, and I find it difficult to decide on the best approach without applying uninformed guesswork regarding the data consumer. 

I don't think the argument of "always choose the generic option" holds here. Choosing 
`https://schema.org/Organization`
over 
`https://service.unece.org/trade/uncefact/vocabulary/uncefact/#Organization`
is straight forward because it's the _exact same_ meaning, and many more will understand the former. For BL# the latter is a _type of_ the former.

**Comment by nissimsan** [comment]
Would it make any sense to allow for multiple/array of term definitions?

**Comment by TallTed** [comment]
A reasoner can be brought to bear such that an `https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Bill_of_lading_number` is _implicitly_ also an `https://schema.org/identifier`.  This is generally optimal.

However, a reasoner is not always usable, for various reasons.  In these cases, it would often if not always be better to multi-type the entity -- that is, an `https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Bill_of_lading_number` is *explicitly* also an `https://schema.org/identifier.

When starting from a specialized ontology of this sort, it is often helpful to include broaderThan, narrowerThan, equivalentProperty, equivalentClass, subClass, subProperty, and similar statements relating the more specialized ontology to the more generic.  These make it viable for following users to use whichever descriptor (a/k/a attribute, predicate, property, etc.) makes sense to them at the time -- that is, it doesn't matter whether they use `https://schema.org/Organization` or `https://service.unece.org/trade/uncefact/vocabulary/uncefact/#Organization` -- but it certainly will have an impact if they use the very broad and generic `https://schema.org/identifier` instead of the very narrow and specific `https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Bill_of_lading_number`.

**Comment by VladimirAlexiev** [comment]
Despite the capitalization, `uncl1153:Bill_of_lading_number` is a property, just like `schema:identifier` is a property.
- @nissimsan: the JSONLD context must map a term `billOfLading` to one or the other, it cannot map it to both
- @TallTed: I agree with you but will put it more strongly: if this project ever provides a "traceability ontology", it should state 
  `uncl1153:Bill_of_lading_number rdfs:subPropertyOf schema:identifier` 
  - you can see here https://schema.org/identifier#subtypes that  `schema:identifier` already has a bunch of sub-properties
  - Even if this ontology is never used with a reasoner, such declaration will be useful and appropriate
- @TallTed: there is entity to "multi-type" here because these are properties not classes

Back to the original question of @nissimsan :
- Since this prop appears in node with type `traceability:BillOfLading`, the prop `schema:identifier` is sufficient to express that is a BL#
- But if the node could have multiple types, or the prop could appear in a differently typed node, then `schema:identifier` is ambiguous
- So I'd prefer the more specific property `uncl1153:Bill_of_lading_number`
- Schema also allows you to use a prefix eg `schema:identifier "BL:12345"` but that prefix "BL" will have no global meaning

**Comment by TallTed** [comment]
@VladimirAlexiev -- 

The kind of a thing a thing is is its type, a/k/a its class. `uncl1153:Bill_of_lading_number` is a property, but each (literal) value of that property (e.g., "12345") is multi-typed as both an `uncl1153:Bill_of_lading_number` and a `schema:identifier`.

(Note that even as you're telling me that `uncl1153:Bill_of_lading_number` is a property which is an `rdfs:subPropertyOf` `schema:identifier`, the page you link to shows the list of `schema:identifier` subProperties in a fragment identified as `subtypes`.)

In other words: very little of this sort of work has been done with clarity for those who didn't participate in doing that work.

**Comment by VladimirAlexiev** [comment]
A value is "typed" by its property? Ok, we obviously won't agree on terminology.

Cheers!

**Comment by nissimsan** [comment]
Thanks @TallTed and @VladimirAlexiev ! 

I will go with `uncl1153:Bill_of_lading_number rdfs:subPropertyOf schema:identifier`

`uncl1153:Bill_of_lading_number rdfs:subPropertyOf schema:identifier` makes a ton of sense, and is precisely expressing what I was aiming for. I have not heard any talks of a "traceability ontology" here, and doubt it will be scoped any time soon. But really, I think this statement belongs on the UN/CEFACT side. I'm involved in that project too and will raise it there. There are many similar situations with CEFACT assuming that there are no other terms in the world by CEFACT's own, and your suggestion here could be a good mechanism to address that.

**Comment by BenjaminMoe** [comment]
I don't disagree with the general statement being put forward. but this issue should be updated with specific change requests or closed.

**Comment by nissimsan** [comment]
This ticket was created to capture and resolve a discussion on a call a while back. 

We generally strive to choose the most generic term definitions. For example, we choose to use schema.org's `Address`, and not CEFACT's `Address`, because we assert that 1) they are meant to define the same thing, and 2) schema.org is more broadly used term and thereby the better choice from an interoperability perspective. 

The issue concerns a nuance to this: a Bill of Lading number can be considered a schema.org `Identifier`, but CEFACT's `Bill_of_lading_number` is a _more precise_ definition of this term, in which case it is the best choice. 

The conversation had basically concluded this already. And it also seemed to be the consensus on the May 24th 2022 call. This ticket seems ripe to close - please add some 👍  in support of closure on this conclusion.

**Comment by OR13** [comment]
I suggest closing this issue, the advice applies to all schemas.
